"git lfs pull origin" hangs indefinitely after upgrade to 2.6.0

```
$ GIT_TRACE=1 git lfs pull origin 
13:37:24.585686 git.c:654               trace: exec: git-lfs pull origin
13:37:24.585782 run-command.c:637       trace: run_command: git-lfs pull origin
13:37:24.598218 trace git-lfs: exec: git 'version'
13:37:24.610754 trace git-lfs: exec: git 'config' '-l'
13:37:24.615020 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
13:37:24.619636 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
13:37:24.628521 trace git-lfs: tq: running as batched queue, batch size of 100
13:37:26.553478 trace git-lfs: filepathfilter: rejecting "AAA/.derived" via []
13:37:26.553571 trace git-lfs: filepathfilter: accepting "AAA/.derived"
13:37:26.553641 trace git-lfs: filepathfilter: rejecting "AAA/.project" via []
...
```
then the process is stuck.  

Current workaround is to downgrade to git-lfs version 2.5.0.

Versions:
```
$ git lfs env
git-lfs/2.6.0 (GitHub; linux amd64; go 1.11.1; git 93702d75)
git version 2.18.0

Endpoint=https://XXXX/info/lfs (auth=basic)
LocalWorkingDir=/home/jenkins/workspace/workspace/AAAA
LocalGitDir=/home/jenkins/workspace/workspace/AAAA/.git
LocalGitStorageDir=/home/jenkins/workspace/workspace/AAAA/.git
LocalMediaDir=/home/jenkins/workspace/workspace/AAAA/.git/lfs/objects
LocalReferenceDirs=
TempDir=/home/jenkins/workspace/workspace/AAAA/.git/lfs/tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=/home/jenkins/workspace/workspace/AAAA/.git/lfs
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic
UploadTransfers=basic
GIT_EXEC_PATH=/usr/libexec/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

> @whitty That's in a pull request ([git-lfs/gitobj#5](https://github.com/git-lfs/gitobj/pull/5)), and it should theoretically be available. I believe the change should be vendored in, so if you check out the repository, it should probably be in `vendor/`, which means that you may already have the change. You may be able to produce a tarball with `git archive` and try building that, which may cause the vendor directory to be used.

I tried the `git-archive` method and unpacked to a separate directory, but that still looks for gitobj upstream (and doesn't behave nicely without being in git - probably `git-describe` is used?)  Perhaps I'm missing something?  I note the "build from source" notes in the wiki don't mention building outside git.

```
$ make
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
go: finding github.com/git-lfs/gitobj v1.0.0-20181107174809-4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e
go: github.com/git-lfs/gitobj@v1.0.0-20181107174809-4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e: unknown revision 4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e
go: error loading module requirements
go: finding github.com/git-lfs/gitobj v1.0.0-20181107174809-4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e
go: github.com/git-lfs/gitobj@v1.0.0-20181107174809-4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e: unknown revision 4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e
go: error loading module requirements
git-lfs: skipping fmt, no goimports found at `goimports` ...
go generate github.com/git-lfs/git-lfs/commands
go: finding github.com/git-lfs/gitobj v1.0.0-20181107174809-4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e
go: github.com/git-lfs/gitobj@v1.0.0-20181107174809-4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e: unknown revision 4be4f67f60bbb94f22c55c1b9e526dfd11e6e07e
go: error loading module requirements
Makefile:160: recipe for target 'commands/mancontent_gen.go' failed
```

> If not, let me know and I can provide a binary.

That would help - at least to verify we have a single problem

--- a/commands/command_status.go
+++ b/commands/command_status.go
@@ -34,8 +34,6 @@ func statusCommand(cmd *cobra.Command, args []string) {
 
 	scanner, err := lfs.NewPointerScanner()
 	if err != nil {
-		scanner.Close()
-
 		ExitWithError(err)
 	}
 

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/git-lfs/git-lfs
 require (
 	github.com/ThomsonReutersEikon/go-ntlm v0.0.0-20151030004737-b00ec39bbdd0
 	github.com/alexbrainman/sspi v0.0.0-20180125232955-4729b3d4d858
-	github.com/git-lfs/gitobj v1.0.0
+	github.com/git-lfs/gitobj v1.1.0
 	github.com/git-lfs/go-netrc v0.0.0-20180525200031-e0e9ca483a18
 	github.com/git-lfs/wildmatch v1.0.0
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect

--- a/lfs/gitscanner_tree.go
+++ b/lfs/gitscanner_tree.go
@@ -49,8 +49,6 @@ func runScanTree(cb GitScannerFoundPointer, ref string, filter *filepathfilter.F
 func catFileBatchTree(treeblobs *TreeBlobChannelWrapper) (*PointerChannelWrapper, error) {
 	scanner, err := NewPointerScanner()
 	if err != nil {
-		scanner.Close()
-
 		return nil, err
 	}
 

--- a/lfs/gitscanner_tree.go
+++ b/lfs/gitscanner_tree.go
@@ -56,8 +56,10 @@ func catFileBatchTree(treeblobs *TreeBlobChannelWrapper) (*PointerChannelWrapper
 	errchan := make(chan error, 10) // Multiple errors possible
 
 	go func() {
+		hasNext := true
 		for t := range treeblobs.Results {
-			hasNext := scanner.Scan(t.Sha1)
+			hasNext = scanner.Scan(t.Sha1)
+
 			if p := scanner.Pointer(); p != nil {
 				p.Name = t.Filename
 				pointers <- p
@@ -72,10 +74,14 @@ func catFileBatchTree(treeblobs *TreeBlobChannelWrapper) (*PointerChannelWrapper
 			}
 		}
 
-		// Deal with nested error from incoming treeblobs
-		err := treeblobs.Wait()
-		if err != nil {
-			errchan <- err
+		// If the scanner quit early, we may still have treeblobs to
+		// read, so waiting for it to close will cause a deadlock.
+		if hasNext {
+			// Deal with nested error from incoming treeblobs
+			err := treeblobs.Wait()
+			if err != nil {
+				errchan <- err
+			}
 		}
 
 		if err = scanner.Close(); err != nil {

--- a/t/t-fetch.sh
+++ b/t/t-fetch.sh
@@ -51,8 +51,9 @@ begin_test "init for fetch tests"
   git push origin newbranch
   assert_server_object "$reponame" "$b_oid"
 
-  # This clone is used for subsequent tests
+  # These clones are used for subsequent tests
   clone_repo "$reponame" clone
+  git clone --shared "$TRASHDIR/clone" "$TRASHDIR/shared"
 )
 end_test
 
@@ -63,6 +64,21 @@ begin_test "fetch"
   rm -rf .git/lfs/objects
 
   git lfs fetch 2>&1 | grep "Downloading LFS objects: 100% (1/1), 1 B"
+  assert_local_object "$contents_oid" 1
+
+  git lfs fsck 2>&1 | tee fsck.log
+  grep "Git LFS fsck OK" fsck.log
+)
+end_test
+
+begin_test "fetch (shared repository)"
+(
+  set -e
+  cd shared
+  rm -rf .git/lfs/objects
+
+  git lfs fetch 2>&1 | tee fetch.log
+  ! grep "Could not scan" fetch.log
   assert_local_object "$contents_oid" 1
 
   git lfs fsck 2>&1 | tee fsck.log

